elements/lisk-transactions/src/sign.ts

```diff
@@ -0,0 +1,182 @@
+		...transactionObject,
+		asset: assetBytes,
+		signatures: [],
+	});
+
+	return transactionBytes;
+};
+
+export const signTransaction = (
+	assetSchema: object,
+	transactionObject: Record<string, unknown>,
+	networkIdentifier: Buffer,
+	senderPassphrase: string,
+): Record<string, unknown> => {
+	if (!networkIdentifier.length) {
+		throw new Error('Network identifier is required to sign a transaction');
+	}
+
+	if (!senderPassphrase) {
+		throw new Error('Passphrase is required to sign a transaction');
+	}
+	validateTransactionSchema(assetSchema, transactionObject);
+	const { publicKey } = getAddressAndPublicKeyFromPassphrase(senderPassphrase);
+
+	if (
+		!Buffer.isBuffer(transactionObject.senderPublicKey) ||
+		!transactionObject.senderPublicKey.equals(publicKey)
+	) {
+		throw new Error('Transaction senderPublicKey does not match public key from passphrase');
+	}
+
+	const transactionWithNetworkIdentifierBytes = Buffer.concat([
+		networkIdentifier,
+		getSigningBytes(assetSchema, transactionObject),
+	]);
+
+	const signature = signData(transactionWithNetworkIdentifierBytes, senderPassphrase);
+	// Reset signatures
+	if (!Array.isArray(transactionObject.signatures)) {
+		throw new Error('Signatures must be of type array');
+	}
+	transactionObject.signatures.push(signature);
+	return transactionObject;
+};
+
+const sanitizeSignaturesArray = (
+	transactionObject: Record<string, unknown>,
+	keys: MultiSignatureKeys,
+	includeSenderSignature: boolean,
+): void => {
+	let numberOfSignatures = keys.mandatoryKeys.length + keys.optionalKeys.length;
+	// Add one extra for multisig account registration
+	if (includeSenderSignature) {
+		numberOfSignatures += 1;
+	}
+
+	for (let i = 0; i < numberOfSignatures; i += 1) {
+		if (
+			Array.isArray(transactionObject.signatures) &&
+			transactionObject.signatures[i] === undefined
+		) {
+			// eslint-disable-next-line no-param-reassign
+			transactionObject.signatures[i] = Buffer.from('');
+		}
+	}
+};
+
+export const signMultiSignatureTransaction = (
+	assetSchema: object,
+	transactionObject: Record<string, unknown>,
+	networkIdentifier: Buffer,
+	senderPassphrase: string,
+	keys: MultiSignatureKeys,
+	includeSenderSignature = false,
+): Record<string, unknown> => {
+	validateTransactionSchema(assetSchema, transactionObject);
+
+	// Sort keys
+	sortKeysAscending(keys.mandatoryKeys);
+	sortKeysAscending(keys.optionalKeys);
+
+	const { publicKey } = getAddressAndPublicKeyFromPassphrase(senderPassphrase);
+
+	const transactionWithNetworkIdentifierBytes = Buffer.concat([
+		networkIdentifier,
+		getSigningBytes(assetSchema, transactionObject),
+	]);
+
+	const signature = signData(transactionWithNetworkIdentifierBytes, senderPassphrase);
+
+	if (
+		includeSenderSignature &&
+		Buffer.isBuffer(transactionObject.senderPublicKey) &&
+		publicKey.equals(transactionObject.senderPublicKey)
+	) {
+		(transactionObject.signatures as Array<Readonly<Buffer>>).unshift(signature);
```

Comment:
What is this doing?
it seems there is no logic in the original code, also `unshift` does not mutate the original array